equinix/fabric_connection_schema.go

```diff
@@ -0,0 +1,840 @@
+package equinix
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var createServiceTokenRes = &schema.Resource{
+	Schema: createServiceTokenSch(),
+}
+
+func createServiceTokenSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"type": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice([]string{"VC_TOKEN"}, true),
+			Description:  "Token type",
+		},
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "An absolute URL that is the subject of the link's context",
+		},
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Equinix-assigned service token identifier",
+		},
+		"description": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Service token description",
+		},
+	}
+}
+
+var createLocationRes = &schema.Resource{
+	Schema: createLocationSch(),
+}
+
+func createLocationSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"region": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Access point region",
+		},
+		"metro_name": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Access point metro name",
+		},
+		"metro_code": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Access point metro code",
+		},
+		"ibx": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "IBX",
+		},
+	}
+}
+
+func createVirtualGatewaySch() map[string]*schema.Schema {
```

Comment:
I'm a little confused why this is the `createVirtualGatewaySch()` function but all of the fields are `Computed`.  Could you have reused `readVirtualGatewaySch` here, or did you intend to make these fields settable?

Reply:
It was an accident, have I tested for Gateway Connection type, it would have failed.  Due to other dependencies I could not test it. I have taken care. Thanks for catching it.